pkg/network/infraconfigurators/masquerade_test.go

```diff
@@ -0,0 +1,148 @@
+	netdriver "kubevirt.io/kubevirt/pkg/network/driver"
+	"kubevirt.io/kubevirt/pkg/virt-launcher/virtwrap/api"
+)
+
+var _ = Describe("Masquerade infrastructure configurator", func() {
+	var (
+		cacheFactory cache.InterfaceCacheFactory
+		ctrl         *gomock.Controller
+		handler      *netdriver.MockNetworkHandler
+	)
+
+	BeforeEach(func() {
+		cacheFactory = fake.NewFakeInMemoryNetworkCacheFactory()
+		ctrl = gomock.NewController(GinkgoT())
+		handler = netdriver.NewMockNetworkHandler(ctrl)
+	})
+
+	AfterEach(func() {
+		ctrl.Finish()
+	})
+
+	const (
+		bridgeIfaceName = "br0"
+	)
+
+	newVMIMasqueradeInterface := func(namespace string, name string) *v1.VirtualMachineInstance {
+		vmi := v1.NewMinimalVMIWithNS(namespace, name)
+		vmi.Spec.Networks = []v1.Network{*v1.DefaultPodNetwork()}
+		vmi.Spec.Domain.Devices.Interfaces = []v1.Interface{*v1.DefaultMasqueradeNetworkInterface()}
+		v1.SetObjectDefaults_VirtualMachineInstance(vmi)
+		return vmi
+	}
+
+	Context("discover link information", func() {
+		const (
+			ifaceName           = "eth0"
+			launcherPID         = 1000
+			ipv4CidrToConfigure = api.DefaultVMCIDR
+			ipv6CidrToConfigure = api.DefaultVMIpv6CIDR
+			ipv4GwAddr          = "10.0.2.1/24"
+			ipv6GwAddr          = "fd10:0:2::1/120"
+			vmIPv4Addr          = "10.0.2.2/24"
+			vmIPv6Addr          = "fd10:0:2::2/120"
+		)
+
+		var (
+			masqueradeConfigurator *MasqueradePodNetworkConfigurator
+			iface                  *v1.Interface
+			podLink                *netlink.GenericLink
+			vmi                    *v1.VirtualMachineInstance
+			ipv4GwIP               *netlink.Addr
+			ipv6GwIP               *netlink.Addr
+			ipv4VmIP               *netlink.Addr
+			ipv6VmIP               *netlink.Addr
+		)
+
+		BeforeEach(func() {
+			iface = v1.DefaultBridgeNetworkInterface()
+			vmi = newVMIMasqueradeInterface("default", "vm1")
+
+			masqueradeConfigurator = NewMasqueradePodNetworkConfigurator(vmi, iface, bridgeIfaceName, nil, ipv4CidrToConfigure, ipv6CidrToConfigure, cacheFactory, launcherPID, handler)
+			ipv4GwIP, _ = netlink.ParseAddr(ipv4GwAddr)
+			ipv6GwIP, _ = netlink.ParseAddr(ipv6GwAddr)
+			ipv4VmIP, _ = netlink.ParseAddr(vmIPv4Addr)
+			ipv6VmIP, _ = netlink.ParseAddr(vmIPv6Addr)
+		})
+
+		When("the pod link is defined", func() {
+			BeforeEach(func() {
+				podLink = &netlink.GenericLink{LinkAttrs: netlink.LinkAttrs{Name: ifaceName, MTU: 1000}}
+				handler.EXPECT().LinkByName(ifaceName).Return(podLink, nil)
+			})
+
+			When("the user wants to use an invalid IPv4 CIDR", func() {
+				BeforeEach(func() {
+					handler.EXPECT().GetHostAndGwAddressesFromCIDR(api.DefaultVMCIDR).Return(ipv4GwAddr, vmIPv4Addr, fmt.Errorf("failed to compute addresses"))
+				})
+
+				It("should fail to read the pod link info", func() {
+					Expect(masqueradeConfigurator.DiscoverPodNetworkInterface(ifaceName)).To(HaveOccurred())
+				})
+			})
+
+			When("the user wants to use an invalid IPv4 CIDR", func() {
```

Comment:
Similar comments to the previous ones at IPv4, also this should be IPv6 not IPv4